### PR TITLE
ODH-1707 usage collector report dropped eventsreports

### PR DESCRIPTION
**Pull request by BenjaminDebotte**
# What is the content of this PR
* New spark task for UsageCollectorReportDroppedEvents (first one 🍾 )
  * Thanks for the great piece of code in NetworkErrorsStbBasicAnalytics that saved me lots of time ;)
* Scripts in tools/ to enable easy build-and-test logic for in-docker development
* Optimization over docker image

### CODE REVIEW

File: src/common/basic_analytics/basic_analytics_processor.py [hunk 1]
```diff
@@ -30,6 +30,7 @@ def _prepare_stream(self, read_stream):
         data_stream = read_stream \
             .select(from_json(read_stream["value"].cast("string"), self._schema).alias("json")) \
             .select("json.*")
+
```
Comment (alimaken) [inline src/common/basic_analytics/basic_analytics_processor.py]: Try not to disturb the files which are not to be modified by your implementation. You can always revert them before a commit.

File: src/common/basic_analytics/aggregations.py [hunk 2]
```diff
@@ -6,8 +6,8 @@
 import re
 from abc import ABCMeta, abstractmethod
 
-from pyspark.sql.functions import lit, avg, count, sum, max, min, col, stddev, expr, regexp_replace
 from pyspark.sql.functions import window, concat_ws, approx_count_distinct, explode, create_map
+from pyspark.sql.functions import lit, avg, count, sum, max, min, col, stddev, expr, regexp_replace
```
Comment (alimaken) [inline src/common/basic_analytics/aggregations.py]: Try not to disturb the files which are not to be modified by your implementation. You can always revert them before a commit.

File: src/applications/basic_analytics/eos_stb/dropped_events_driver.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+"""
+The module for the driver to calculate metrics related to Think Analytics HTTP access component.
+"""
+from pyspark.sql.types import StructField, StructType, TimestampType, StringType, LongType
+from pyspark.sql.functions import *
```
Comment (alimaken) [inline src/applications/basic_analytics/eos_stb/dropped_events_driver.py]: Avoid star imports.

File: src/applications/basic_analytics/eos_stb/dropped_events_driver.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+from common.basic_analytics.basic_analytics_processor import BasicAnalyticsProcessor
+from common.basic_analytics.aggregations import *
```
Comment (alimaken) [inline src/applications/basic_analytics/eos_stb/dropped_events_driver.py]: Avoid star imports

File: src/applications/basic_analytics/eos_stb/dropped_events_driver.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+        stream = json_stream\
+            .withColumn("UsageCollectorReport_missed_events", col("UsageCollectorReport_missed_events").cast(LongType()))\
+            .where("UsageCollectorReport_missed_events is not null")
+
+
+        aggregation_field = "UsageCollectorReport_missed_events"
+        result = []
+
+        for field in aggregation_fields:
```
Comment (alimaken) [inline src/applications/basic_analytics/eos_stb/dropped_events_driver.py]: aggregation_fields is not even defined in the context.

Comment (alimaken) [reply src/applications/basic_analytics/eos_stb/dropped_events_driver.py]: You don't need a for loop for this probably.

File: src/applications/basic_analytics/eos_stb/dropped_events_driver.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+            aggregations = [Sum(**kwargs), Count(**kwargs), Max(**kwargs), Min(**kwargs), Stddev(**kwargs),
+                            P01(**kwargs), P05(**kwargs), P10(**kwargs), P25(**kwargs), P50(**kwargs),
+                            P75(**kwargs), P90(**kwargs), P95(**kwargs), P99(**kwargs)]
+
+            result.append(stream.aggregate(CompoundAggregation(aggregations=aggregations, **kwargs)))
+
+        return result
+
+    @staticmethod
+    def create_schema():
```
Comment (alimaken) [inline src/applications/basic_analytics/eos_stb/dropped_events_driver.py]: Public methods must have a doc-string.

File: src/applications/basic_analytics/eos_stb/dropped_events_driver.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
+"""
+The module for the driver to calculate metrics related to Think Analytics HTTP access component.
```
Comment (alimaken) [inline src/applications/basic_analytics/eos_stb/dropped_events_driver.py]: Update the comment according to your job description.

File: configuration/basic_analytics/eos_stb/usagecollector_dropped_events.yml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  master: local[*]
+  checkpointLocation: file:///spark/checkpoints/procs/
+  shuffle.partitions: 1
+  consoleWriter: false
+  outputMode: update
+
+kafka:
+  maxOffsetsPerTrigger: 100000
+  bootstrap.servers: 172.31.101.138:9094,172.31.101.139:9094,172.31.101.140:9094
+  startingOffsets: earliest
```
Comment (alimaken) [inline configuration/basic_analytics/eos_stb/usagecollector_dropped_events.yml]: Should always use `latest` unless explicitly required to use `earliest`.

File: src/applications/basic_analytics/eos_stb/dropped_events_driver.py (hunk 5 above)
Comment (BenjaminDebotte) [reply src/applications/basic_analytics/eos_stb/dropped_events_driver.py]: I've forgot to remove it, I've been a bit too fast on this

**Pull request merged**